packages/framework/fluid-static/src/fluidContainer.ts

```diff
@@ -230,6 +230,20 @@ export interface IFluidContainer<TContainerSchema extends ContainerSchema = Cont
 	dispose(): void;
 }
 
+/**
+ * Internal interface for {@link IFluidContainer}.
+ *
+ * @internal
+ */
+export interface IFluidContainerInternal {
+	/**
+	 * Gets the underlying {@link @fluidframework/container-definitions#IContainer}.
+	 *
+	 * @remarks Used to power debug tooling and experimental features.
+	 */
+	INTERNAL_CONTAINER_DO_NOT_USE?(): IContainer;
```

Comment:
Since this is internal only, we should probably rename it now. And since this was only optional for pre-internal backwards compat, we should make this required.
```suggestion
	readonly container: IContainer;
```

Reply:
I think this one is a little sneaky.
`IFluidContainer` is now `@sealed` which I think means we can assume that if we get an `IFluidContainer` then we expect it to come from use and thus have `IFluidContainerInternal`. So, if we had an alternate implementation that didn't also support `IFluidContainerInternal` that would be on us fully.
Do we all agree on that thinking? (+ @CraigMacomber )

If we keep it as is, then we are free to have `IFluidContainer` that does not have `IFluidContainerInternal`. The current pattern is very safe as the method is optional and forces the user to do a check even after casting.

If we change, I guess we do something like introduce:
```typescript
function toInternalFluidContainer<TContainerSchema extends ContainerSchema>(
	container: IFluidContainer<TContainerSchema> | IFluidContainerInternal
): IFluidContainerInternal {
	if ("INTERNAL_CONTAINER_DO_NOT_USE" in container) {
		return container;
	}
	throw new Error("container is not IFluidContainerInternal");
}
```
But does devtools want to be that strict? Currently it logs console error and moves on.

So, perhaps:
```typescript
/**
 * Internal interface for {@link IFluidContainer}.
 * @internal
 */
export interface IFluidContainerInternal {
	/**
	 * The underlying {@link @fluidframework/container-definitions#IContainer}.
	 * @remarks Used to power debug tooling and experimental features.
	 */
	readonly container: IContainer;
}

/**
 * Check that the provided `container` is an internal {@link IFluidContainerInternal}.
 * @internal
 */
export function isInternalFluidContainer<TContainerSchema extends ContainerSchema>(
	container: IFluidContainer<TContainerSchema> | IFluidContainerInternal,
): container is IFluidContainerInternal {
	return "container" in container;
}
```
and devtools would look like:
```typescript
function mapContainerProps(
	containerProps: ContainerDevtoolsProps,
): ContainerDevtoolsPropsBase | undefined {
	const { container, containerKey } = containerProps;
	if (!isInternalFluidContainer(container)) {
		console.error("Missing Container accessor on FluidContainer.");
		return undefined;
	}

	return {
		container: container.container,
		containerKey,
		containerData: container.initialObjects as Record<string, IFluidLoadable>,
	};
}
```

Reply:
I think it's fine for devtools to be strict about the presence of the internal container. If users want to consumer our devtools, then they must follow our documented contracts and not introduce their own `IFluidContainer` implementations.

Reply:
Updated to failing faster - left the internal utility still "relaxed"; caller decides how to handle